src/meili/match_route.cc

```diff
@@ -147,8 +148,15 @@ void MergeEdgeSegments(std::vector<EdgeSegment>& route,
 namespace valhalla {
 namespace meili {
 
-EdgeSegment::EdgeSegment(baldr::GraphId the_edgeid, float the_source, float the_target)
-    : edgeid(the_edgeid), source(the_source), target(the_target) {
+EdgeSegment::EdgeSegment(baldr::GraphId the_edgeid,
```

Comment:
If I recall correctly, `EdgeSegment` was a segment of an edge from `GraphReader`, with the endpoints truncated by some fraction at the start and the end. is that right?

What do these new indices correspond to?

Reply:
the first match result index is the index of the first tracepoint in the users input that corresponds to the source (truncated from the beginning of the edge) whereas the last match result index is the index of the last tracepoint in the users input that corresponds to the target (truncated from the beginning of the edge). its a way to look at a segment and say, aha, that tracepoint maps to this point on the edge segment. we need it so we chop up legs of a route where user input points are marked as break or break_through